File: app/src/main/java/com/github/overpass/gather/screen/dialog/ProgressDialogFragment.java
package com.github.overpass.gather.screen.dialog;

import android.app.Dialog;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.github.overpass.gather.R;
import com.github.overpass.gather.screen.base.BaseDialogFragment;

public class ProgressDialogFragment extends BaseDialogFragment {

    private static final String TAG = "ProgressDialogFragment";

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setView(R.layout.fragment_dialog_progress)
                .setCancelable(false)
                .create();
    }

    public static void show(@Nullable FragmentManager fragmentManager) {
        show(TAG, fragmentManager, false, ProgressDialogFragment::new);
    }

    public static void hide(@Nullable FragmentManager fragmentManager) {
        hide(TAG, fragmentManager, ProgressDialogFragment.class);
    }
}

File: app/src/main/java/com/github/overpass/gather/screen/dialog/progress/indeterminate/ProgressDialogFragment.kt
package com.github.overpass.gather.screen.dialog.progress.indeterminate

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.FragmentManager

import com.github.overpass.gather.R
import com.github.overpass.gather.screen.base.BaseDialogFragment

class ProgressDialogFragment : BaseDialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return AlertDialog.Builder(context!!)
                .setView(R.layout.fragment_dialog_progress)
                .setCancelable(false)
                .create()
    }

    companion object {

        private const val TAG = "ProgressDialogFragment"

        @JvmStatic
        fun show(fragmentManager: FragmentManager?) {
            show(TAG, fragmentManager, false) { ProgressDialogFragment() }
        }

        @JvmStatic
        fun hide(fragmentManager: FragmentManager?) {
            hide(TAG, fragmentManager, ProgressDialogFragment::class.java)
        }
    }
}
